standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -93,6 +93,11 @@ public CompactionTxnHandler() {
       try {
         dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
         stmt = dbConn.createStatement();
+        // De-duplicate
```

Comment:
Shouldn't we check that the corresponding transaction is not aborted, or open?

Reply:
COMPLETED_TXN_COMPONENTS contains only committed txns. Aborted & open stay in TXN_COMPONENTS

Reply:
Got it. Thanks!